packages/engine-core/src/library/types/Library.ts

```diff
@@ -0,0 +1,50 @@
+import {
+  GetConfigOptions,
+  ConfigMetaFormat,
+  QueryEngineConfig,
+} from '../../common/types/QueryEngine'
+
+export type ConnectArgs = {
+  enableRawQueries: boolean
+}
+
+export type QueryEngineInstance = {
+  connect(connectArgs: ConnectArgs): Promise<void>
+  disconnect(): Promise<void>
+  /**
+   *
+   * @param request JSON.stringified `QueryEngineRequest | QueryEngineBatchRequest`
+   *
+   * @param headers JSON.stringified `QueryEngineRequestHeaders`
+   */
+  query(
+    request: string,
+    headers: string,
```

Comment:
Can we rename `headers`? To `headerStr` maybe. 

headers everywhere else is a `Record<string, string>` so from first look it's confusing